deis/deis
# Use Real CoreOS on DigitalOcean

Note: this won't work. DigitalOceans CoreOS is too new for current deis version...

Howto Test Provisioning:
- edit contrib/coreos/user-data as usual to include your new discovery URL
- `gem install docl`
- `cd contrib/digitalocean`
- Create a personal Access Token on DO (https://cloud.digitalocean.com/settings/applications)
- Copy the token a new file to do-access-token (in contrib/digitalocean)
- exectue ./provision-do-cluster.sh lon1 230790 2GB
  Replace the middle value with your SSH ID
  The first value is identifier for the region (valid values are sfo1 lon1 ny3).
  Last value is the size identifier in the form of NGB (eg 8GB which used to be the default)
- Wait & look for the IP on the DO website in your droplet list
- `ssh core@$PUBLICIP`
- This show show welcome to DEIS
  $ `fleetctl list-machines
  Should list the 3 machines provided (or more if you modified DEIS_NUM_INSTANCES)

TDOO:
- [x] Update DEIS itself so it works with this newer version of CoreOS
- [x] Show progress for creating the Droplets
- [x] Handle errors in creating the droplets
- [x] Decide if we want users to install 2 depencencies (barge DO Library, and tugboat command line), or convert everything to barge (meaning scripts for all possible actions need to be written).

Earlier TODO
- [ ] Start using the Beta channel as soon as possible.
  Apparantly we rely on the alpha channel for all other providers as well, so this todo is discarded now
